common/setopt.c

```diff
@@ -175,3 +175,214 @@ GetOptionAndValue(
     }
     goto cleanup;
 }
+
+static
+uint32_t
+_TDNFGetCmdOpt(
+    PTDNF_CMD_ARGS pArgs,
+    const char *pszOptName,
+    PTDNF_CMD_OPT *ppOpt
+    )
+{
+    uint32_t dwError = 0;
+    PTDNF_CMD_OPT pOpt = NULL;
+    int nHasOpt = 0;
+
+    for (pOpt = pArgs->pSetOpt;
+         pOpt;
+         pOpt = pOpt->pNext)
+    {
+        if (pOpt->nType == CMDOPT_KEYVALUE &&
+            !strcmp(pOpt->pszOptName, pszOptName))
+        {
+            nHasOpt = 1;
+            break;
+        }
+    }
+
+    if (!nHasOpt)
+    {
+        dwError = ERROR_TDNF_FILE_NOT_FOUND;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    *ppOpt = pOpt;
+error:
+    return dwError;
+}
+
+uint32_t
+TDNFHasOpt(
+    PTDNF_CMD_ARGS pArgs,
+    const char *pszOptName,
+    int *pnHasOpt
+    )
+{
+    uint32_t dwError = 0;
+    PTDNF_CMD_OPT pOpt = NULL;
+    int nHasOpt = 0;
+
+    if(!pArgs || !pArgs->pSetOpt ||
+       IsNullOrEmptyString(pszOptName) || !pnHasOpt)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = _TDNFGetCmdOpt(pArgs, pszOptName, &pOpt);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    *pnHasOpt = nHasOpt;
```

Comment:
This always assigns 0 to * pnHasOpt. Is that what is expected here?

Reply:
Good catch. fixed.